### PR TITLE
Makefile: Fix getting misspell code when already installed

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
Grouping misspell installation code so it won't executed when
misspell is found in the current system.

## Motivation and Context
Fix Makefile

## Regression
No

## How Has This Been Tested?
`make test`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by harshavardhana — DISMISSED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#80646f13-608a-40ec-a59e-fca82e126732) |
| mint-compression-dist-xl.sh | :x: [more...](#218f34b8-38ee-4850-a67a-83965283e064) |
| mint-dist-xl.sh | :x: [more...](#48560c6c-2e6e-415e-9ca1-3706ea0572ff) |
<a name="80646f13-608a-40ec-a59e-fca82e126732"></a>
### 7434-b11df00/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:31802
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp db025777efdc:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-jvvndrgwgxisft",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 360016,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 158FE3DA448E4DAE, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="218f34b8-38ee-4850-a67a-83965283e064"></a>
### 7434-b11df00/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:31882
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 38f70878604a:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 4 minutes and 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 46 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 26 seconds
(8/14) Running minio-dotnet tests ... done in 39 seconds
(9/14) Running minio-go tests ... done in 43 seconds
(10/14) Running minio-java tests ... FAILED in 49 seconds
{
  "name": "minio-java",
  "function": "listIncompleteUploads(String bucketName)",
  "duration": 253,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=InternalError, message=We encountered an internal error, please try again., bucketName=minio-java-test-2tjphal, objectName=null, resource=/minio-java-test-2tjphal, requestId=158FE4068045246F, hostId=521252f2-0611-4f49-bf46-d9774041f4de)\nrequest={method=GET, url=http://72.28.97.53:31882/minio-java-test-2tjphal?max-uploads=1000&prefix=&delimiter=&uploads=, headers=Host: 72.28.97.53:31882\nUser-Agent: Minio (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20190327T182513Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20190327/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=500, headers=Accept-Ranges: bytes\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: Minio/DEVELOPMENT.2019-03-27T18-10-30Z\nVary: Origin\nX-Amz-Request-Id: 158FE4068045246F\nX-Minio-Deployment-Id: 521252f2-0611-4f49-bf46-d9774041f4de\nX-Xss-Protection: 1; mode=block\nDate: Wed, 27 Mar 2019 18:25:13 GMT\nTransfer-Encoding: chunked\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1190), io.minio.MinioClient.execute(MinioClient.java:1056), io.minio.MinioClient.executeGet(MinioClient.java:1284), io.minio.MinioClient.listIncompleteUploads(MinioClient.java:4696), io.minio.MinioClient.access$300(MinioClient.java:152), io.minio.MinioClient$6$1.populate(MinioClient.java:4546), io.minio.MinioClient$6$1.hasNext(MinioClient.java:4581), FunctionalTest.listIncompleteUploads_test1(FunctionalTest.java:1614), FunctionalTest.runTests(FunctionalTest.java:2991), FunctionalTest.main(FunctionalTest.java:3121)]"
}

Executed 9 out of 14 tests successfully.
```
<a name="48560c6c-2e6e-415e-9ca1-3706ea0572ff"></a>
### 7434-b11df00/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:31287
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3017756dccb4:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-sygueppe4sufsd",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 297,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.52:31287/aws-sdk-go-test-sygueppe4sufsd: dial tcp 72.28.97.52:31287: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

File: Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@ getdeps:
 	@mkdir -p $(GOPATH)/bin
 	@echo "Installing golint" && which golint || go get -u golang.org/x/lint/golint
 	@echo "Installing staticcheck" && which staticcheck || wget --quiet -O $(GOPATH)/bin/staticcheck https://github.com/dominikh/go-tools/releases/download/2019.1/staticcheck_linux_amd64 && chmod +x $(GOPATH)/bin/staticcheck
-	@echo "Installing misspell" && which misspell || wget --quiet https://github.com/client9/misspell/releases/download/v0.3.4/misspell_0.3.4_linux_64bit.tar.gz && tar xvf misspell_0.3.4_linux_64bit.tar.gz && mv misspell $(GOPATH)/bin/misspell && chmod +x $(GOPATH)/bin/misspell && rm -r misspell_0.3.4_linux_64bit.tar.gz
+	@echo "Installing misspell" && which misspell || { wget --quiet https://github.com/client9/misspell/releases/download/v0.3.4/misspell_0.3.4_linux_64bit.tar.gz && tar xvf misspell_0.3.4_linux_64bit.tar.gz && mv misspell $(GOPATH)/bin/misspell && chmod +x $(GOPATH)/bin/misspell && rm -r misspell_0.3.4_linux_64bit.tar.gz ; }
```
Comment (kannappanr) [inline Makefile]: This doesn't work on `csh` 
```
$ csh
% which misspell || { echo "Intell" && echo "iii" ;}
/home/kannappan/gopath/bin/misspell
}: Command not found.
% which misspell || { echo "Intell" && echo "iii" }     
/home/kannappan/gopath/bin/misspell
% 
```

This seems to work on all the shells

```
@echo "Installing misspell" && which misspell || (wget --quiet https://github.com/client9/misspell/releases/download/v0.3.4/misspell_0.3.4_linux_64bit.tar.gz && tar xvf misspell_0.3.4_linux_64bit.tar.gz && mv misspell ${GOPATH}/bin/misspell && chmod +x ${GOPATH}/bin/misspell && rm -r misspell_0.3.4_linux_64bit.tar.gz)
```

File: Makefile [hunk 2]
```diff
@@ -16,7 +16,7 @@ getdeps:
 	@mkdir -p $(GOPATH)/bin
 	@echo "Installing golint" && which golint || go get -u golang.org/x/lint/golint
 	@echo "Installing staticcheck" && which staticcheck || wget --quiet -O $(GOPATH)/bin/staticcheck https://github.com/dominikh/go-tools/releases/download/2019.1/staticcheck_linux_amd64 && chmod +x $(GOPATH)/bin/staticcheck
```
Comment (kannappanr) [inline Makefile]: Why not do the same for `staticcheck` also?

**Comment by harshavardhana** [comment]
I have this fixed in go1.12 migration PR this needs to be added

```
getdeps:
        @mkdir -p $(GOPATH)/bin
        @which golint 1>/dev/null || (echo "Installing golint" && go get -u golang.org/x/lint/golint)
        @which staticcheck 1>/dev/null || (echo "Installing staticcheck" && wget --quiet -O $(GOPATH)/bin/staticcheck https://github.com/dominikh/go-tools/releases/download/2019.1/staticcheck_linux_amd64 && chmod +x $(GOPATH)/bin/staticcheck)
        @which misspell 1>/dev/null || (echo "Installing misspell" && wget --quiet https://github.com/client9/misspell/releases/download/v0.3.4/misspell_0.3.4_linux_64bit.tar.gz && tar xf misspell_0.3.4_linux_64bit.tar.gz && mv misspell $(GOPATH)/bin/misspell && chmod +x $(GOPATH)/bin/misspell && rm -f misspell_0.3.4_linux_64bit.tar.gz)
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#354ea21a-cfc7-49a8-bff3-7119bb4e1abb) |
| mint-worm.sh | :x: [more...](#b66ce531-2ba7-4cab-80fc-6826141032f0) |
<a name="b66ce531-2ba7-4cab-80fc-6826141032f0"></a>
### 7434-740333d/mint-worm.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30707
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            worm
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a37ed5e59b96:/mint/log /tmp/mint-logs'

(1/1) Running worm tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "bucket-jn4ok7mzn04c0xul94rwoaflyrs5azdvynd3jtan5vsj5zxj08fhflh",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 277,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.56:30707/bucket-jn4ok7mzn04c0xul94rwoaflyrs5azdvynd3jtan5vsj5zxj08fhflh: dial tcp 72.28.97.56:30707: connect: connection refused",
  "function": "PutAndDelete",
  "message": "WORM_MODE ON  - CreateBucket Failed",
  "name": "test worm mode",
  "status": "FAIL"
}

Executed 0 out of 1 tests successfully.
```
<a name="354ea21a-cfc7-49a8-bff3-7119bb4e1abb"></a>
### 7434-740333d/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:32634
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4fa287f66a96:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-l4vgh41z0hsqne",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 278,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.55:32634/aws-sdk-go-test-l4vgh41z0hsqne: dial tcp 72.28.97.55:32634: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#6c38f75f-a31a-48bd-85de-32d35f2d2f64) |
| mint-large-bucket.sh | :x: [more...](#25f6b7d3-90a6-4ef1-9bdd-28a9fa04eeae) |
| mint-compression-dist-xl.sh | :x: [more...](#d5e12253-934b-4918-a7b2-095652c8b15c) |
| mint-dist-xl.sh | :x: [more...](#42d806a6-04d5-44fd-997d-0c5f3e83551f) |
<a name="6c38f75f-a31a-48bd-85de-32d35f2d2f64"></a>
### 7434-7c93dd4/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:31496
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b82b70d56cf7:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-xypxoj6q03464n",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 346,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.53:31496/aws-sdk-go-test-xypxoj6q03464n: dial tcp 72.28.97.53:31496: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="42d806a6-04d5-44fd-997d-0c5f3e83551f"></a>
### 7434-7c93dd4/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:30540
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ae4fb3d16d13:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-nu0u2mdkjarnqp",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 326,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.60:30540/aws-sdk-go-test-nu0u2mdkjarnqp: dial tcp 72.28.97.60:30540: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="25f6b7d3-90a6-4ef1-9bdd-28a9fa04eeae"></a>
### 7434-7c93dd4/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30539
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a61d9c299d68:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-devm9vqn529yiu",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 347,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.56:30539/aws-sdk-go-test-devm9vqn529yiu: dial tcp 72.28.97.56:30539: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="d5e12253-934b-4918-a7b2-095652c8b15c"></a>
### 7434-7c93dd4/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:31064
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8e348b09222b:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-qnujekkudhykff",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 406,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.60:31064/aws-sdk-go-test-qnujekkudhykff: dial tcp 72.28.97.60:31064: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Pull request merged**